### PR TITLE
go.mod: fix a cve in a transitive dependency to make Dependabot happy 

### PR DESCRIPTION
**Pull request by mpimenov**
Dependabot found a bug in a transitive dependency of one of our tests.
Fixing it here by bumping the version.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1459,8 +1459,9 @@ gopkg.in/yaml.v2 v2.3.0/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
 gopkg.in/yaml.v2 v2.4.0 h1:D8xgwECY7CYvx+Y2n4sBz93Jn9JRvxdiyyo8CTfuKaY=
 gopkg.in/yaml.v2 v2.4.0/go.mod h1:RDklbk79AGWmwhnvt/jBztapEOGDOx6ZbXqjP6csGnQ=
 gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c/go.mod h1:K4uyk7z7BCEPqu6E+C64Yfv1cQ7kz7rIZviUmN+EgEM=
-gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b h1:h8qDotaEPuJATrMmW04NCwg7v22aHH28wwpauUhK9Oo=
 gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b/go.mod h1:K4uyk7z7BCEPqu6E+C64Yfv1cQ7kz7rIZviUmN+EgEM=
```
Comment (mpimenov) [inline go.sum:1462]: Why is this not removed?

File: go.mod [hunk 2]
```diff
@@ -92,6 +92,7 @@ require (
 	google.golang.org/grpc v1.46.0 // indirect
 	google.golang.org/protobuf v1.28.0 // indirect
 	gopkg.in/warnings.v0 v0.1.2 // indirect
+	gopkg.in/yaml.v3 v3.0.0 // indirect
```
Comment (mpimenov) [inline go.mod:95]: I don't like that we now have this in our go.mod.

Comment (keegancsmith) [reply go.mod:95]: if it fixes the warning seems fine

File: go.sum (hunk 1 above)
Comment (keegancsmith) [reply go.sum:1462]: go.sum doesn't record what is used in the final state, but all intermediate states that are checked. so likely this version is in a dependencies go.mod.

**Review by keegancsmith — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (mpimenov) [reply go.mod:95]: Let's see if it does.

**Pull request merged**